src/main/java/com/linecorp/armeria/client/routing/RoundRobinStrategy.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.math.IntMath;
+
+import java.util.List;
+
+/**
+ * A weighted round robin select strategy.
+ * <p>
+ * For nodes a,b,c<br>
+ * if server node weights are 1,1,1(or 2,2,2),then select result is abc abc ...<br>
+ * if server node weights are 1,2,3(or 2,4,6),then select result is abbccc abbccc ...<br>
+ * if server node weights are 3,5,7,then select result is aaabbbbbccccccc aaabbbbbccccccc ...
+ */
+public class RoundRobinStrategy implements NodeSelectStrategy {
```

Comment:
I guess we can still avoid the synchronization with some pre-populated weight-compensated node list, though.

The real question here would be: should 'weight' be an integral part of a server group? I'm sure simple round-robin strategy will not need weight at all as you said, and some interesting strategy implementation might even use its own exotic properties.